REVIEWING.md

```diff
@@ -0,0 +1,125 @@
+# Reviewing PRs at TensorZero
+
+We require a code review of every PR to the repository.
+Code review itself is an important part of the engineering process and should allow us to catch issues with correctness, developer experience, and conformance to a spec.
+In this document, we list some guidelines (a checklist) to follow when reviewing PRs so that we do not forget to check the necessary boxes.
+We omit changes that will be automatically caught by CI.
+
+## Reviewer checklist
+
+### General
+
+* Does the PR have a clear purpose? Could it be split into more focused changes?
+* Does the PR add any dependencies? Is there a good justification for adding them? Are we sure we want to depend on them?
+
+### Specification
+
+* Was a spec written for the change?
```

Comment:
Let's figure out a way to include specs/design docs in the codebase itself so it's easy to reference them, possibly with golinks or something similar with access control? Otherwise, we could design in the open but that might slow us down.

Reply:
We should have more specs in Markdown in the codebase so that agents can easily reference them.